new.go

```diff
@@ -166,6 +166,12 @@ func resolveImage(ctx context.Context, systemContext *types.SystemContext, store
 		}
 	}
 
+	// If we found a local image, we must use it's name.
+	// See #2904.
+	if localImageRef != nil {
+		fromImage = localImageName
```

Comment:
(Non-blocking: It might be useful to add a debug log if `fromImage != localImageName`, so that users can figure out where the fully-qualified variant came from.)